test/revolution_test.rb

```diff
@@ -0,0 +1,522 @@
+            ["cameraHead"]["camera"]["brightness"], \
+                     json_from_expected["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["brightness"]
+        assert_equal json_from_sample["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["focus"], \
+                     json_from_expected["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["focus"]
+        assert_equal json_from_sample["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["saturation"], \
+                     json_from_expected["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["saturation"]
+        assert_equal json_from_sample["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["sharpness"], \
+                     json_from_expected["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["sharpness"]
+        assert_equal json_from_sample["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["zoom"]["ratio"], \
+                     json_from_expected["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["zoom"]["ratio"]
+        assert_equal json_from_sample["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["zoom"]["speed"], \
+                     json_from_expected["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["zoom"]["speed"]
+    end
+
+    it "sends grabber command" do
+        raw_cmd = raw_packet_input
+        # cmd action input
+        cmd_action = Types.deep_trekker.DevicesCommandAction
+        cmd_action = "GrabberCommandAction"
+        # grabber cmd input
+        cmd = Types.deep_trekker.GrabberCommand.new
+        cmd.open = 1
+        cmd.speed = Types.base.samples.Joints.new(
+            time: Time.now,
+            elements: [
+                Types.base.JointState.Speed(3.0)
+            ]
+        )
+
+        sample = expect_execution do
+            syskit_write task.data_in_port, raw_cmd
+            syskit_write task.command_action_port, cmd_action
+            syskit_write task.grabber_command_port, cmd
+        end.to do
+            have_one_new_sample task.data_out_port
+        end
+
+        expected_raw_cmd = expected_grabber_raw_data(cmd)
+
+        json_from_sample = JSON.parse(sample.data.to_byte_array[8..-1])
+        json_from_expected = JSON.parse(expected_raw_cmd.data.to_byte_array[8..-1])
+
+        assert_equal json_from_sample["payload"]["devices"]["57B974C0A269"] \
+            ["grabber"]["openClose"], json_from_expected["payload"]["devices"] \
+            ["57B974C0A269"]["grabber"]["openClose"]
+        assert_equal json_from_sample["payload"]["devices"]["57B974C0A269"] \
+            ["grabber"]["rotate"], json_from_expected["payload"]["devices"] \
+            ["57B974C0A269"]["grabber"]["rotate"]
```

Comment:
Even tests with few assertions like this one would improve from transforming the json access into variables.